ui/src/components/HeaderBar.tsx

```diff
@@ -0,0 +1,55 @@
+import { Box, Flex, HStack, IconButton, Input, Link, Text, useColorModeValue } from "@chakra-ui/react"
+import React from "react"
+import { FaGithub } from "react-icons/fa"
+import ColorToggleButton from "./ColorToggleButton"
+
+function HeaderBar({
+  onQueryChange,
+  error
+}: Readonly<{
+  onQueryChange: (query: string) => void
+  error: boolean
+}>): React.ReactElement {
+  return (
+    <Box
+      as="header"
+      position="sticky"
+      top="0"
+      zIndex="1000"
+      bg={useColorModeValue("gray.100", "gray.900")}
+      boxShadow="sm"
+      padding={4}
+    >
+      <Flex justifyContent="space-between" alignItems="center">
+        {/* Left: Icon and Project Name */}
+        <HStack>
+          <Text fontSize="xl" fontWeight="bold">
+            SwarmCD
+          </Text>
+        </HStack>
+
+        {/* Middle: Search Bar */}
+        <Box flex="1" mx={6}>
+          <Input
+            placeholder="Search..."
+            onChange={event => onQueryChange(event.target.value)}
+            size="lg"
+            variant="filled"
+            bg={useColorModeValue("gray.200", "gray.800")}
+            disabled={error}
+          />
+        </Box>
```

Comment:
Could be extracted into a private function in this class. IMO that should be done for all the blocks here instead of separating them with comments.

Reply:
Done!